pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -147,11 +147,16 @@ static int _open(vfs_file_t *filp, const char *name, int flags, mode_t mode,
     if ((flags & O_APPEND) == O_APPEND) {
         fatfs_flags |= FA_OPEN_APPEND;
     }
-    if ((flags & O_TRUNC) == O_TRUNC) {
-        fatfs_flags |= FA_CREATE_ALWAYS;
-    }
```

Comment:
> If the file already exists and is a regular file and the access mode allows writing (i.e., is O_RDWR or O_WRONLY) it will be truncated to length 0.
> […] Otherwise, the effect of O_TRUNC is unspecified.

So no need to move this. That needlessly changes the behavior of existing code that used to work with only `O_TRUNC`.

Reply:
Ok, sounds reasonable.